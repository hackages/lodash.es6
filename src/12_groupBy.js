/**
 * Creates an object composed of keys generated from the results of running each element of collection thru iteratee
 */
export function groupBy(collection, iteratee) {}
