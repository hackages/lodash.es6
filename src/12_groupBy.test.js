import {groupBy} from './12_groupBy'

describe('groupBy', () => {
  it('should group', () => {
    expect(
      groupBy(
        [
          {sex: 'M', name: 'Robert'},
          {sex: 'M', name: 'Bob'},
          {sex: 'F', name: 'Louise'},
        ],
        e => e.sex
      )
    ).toMatchSnapshot()
  })
})
