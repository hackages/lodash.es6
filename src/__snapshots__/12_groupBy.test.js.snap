// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupBy should group 1`] = `
Object {
  "F": Array [
    Object {
      "name": "Louise",
      "sex": "F",
    },
  ],
  "M": Array [
    Object {
      "name": "Robert",
      "sex": "M",
    },
    Object {
      "name": "Bob",
      "sex": "M",
    },
  ],
}
`;
